README.rst

|PypiVersion| |Doc badge| |Travis| |Pypi|

===============
Django-Leonardo
===============

Full featured platform for fast and easy building extensible web applications.

*Don't waste your time searching stable solution for daily problems.*

**Deploy and Enjoy ! No skills required !**

For Users
=========

* CMS, Page, Responsive, Layouts, Themes, Color Variations 
* Widgets, Plugins, 3rd party app integrations
* Frontend Edit, Install/Uninstall modules in one click !
* Store, Form Designer, Blog, News, Folio, Links, Navigations, ..
* Media, Folders, Files, Images, Documents, Import - Export, ..
* LIVE settings, Auto loading modules, editable templates, ..
* Authentification, 3rd party backends, SAML standard, ..

For Developers
==============

* Python, Django, FeinCMS, OpenStack Horizon
* AngularJS, React, Bootstrap, Compress, Bootswatch
* Crispy forms, Floppy forms, Select2
* Filer, DbTemplates, Reversion, Constance
* Haystack, Oscar, Django Admin Bootstrap
* and tons of other apps bundled as modules

Further reading:

* `Leonardo Site`_
* `Demo Site`_ (a reference build of an Leonardo project)
* `Demo Store`_ (a reference build of an Leonardo Store project)
* `Developer Documentation`_ (documentation for Django Developers)
* `User Documentation`_ (documentation for Leonardo end-users)
* `Modules`_ Leonardo modules
* `Continuous integration homepage`_

.. image:: https://badges.gitter.im/Join%20Chat.svg
   :alt: Join the chat at https://gitter.im/django-leonardo/django-leonardo
   :target: https://gitter.im/django-leonardo/django-leonardo?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge

.. image:: https://coveralls.io/repos/django-leonardo/django-leonardo/badge.svg?branch=develop
   :alt: Coverage
   :target: https://coveralls.io/r/django-leonardo/django-leonardo?branch=develop

.. _`Leonardo Site`: http://www.leonardo-cms.org
.. _`Demo Site`: http://demo.leonardo-cms.org
.. _`Modules`: http://github.com/leonardo-modules
.. _`Demo Store`: http://store.leonardo-cms.org
.. _`Continuous integration homepage`: http://travis-ci.org/django-leonardo/django-leonardo
.. _`Developer Documentation`: http://django-leonardo.readthedocs.org
.. _`User Documentation`: http://leonardo-documentation.rtfd.org

Installation
============

one liner

Wget

.. code-block:: bash

    wget -O - https://github.com/django-leonardo/django-leonardo/raw/develop/contrib/scripts/install_leonardo.sh | sh


Install Blog

.. code-block:: bash

    wget -O - https://github.com/django-leonardo/django-leonardo/raw/develop/contrib/scripts/install_blog.sh | sh

Install Store

.. code-block:: bash

    wget -O - https://github.com/django-leonardo/django-leonardo/raw/develop/contrib/scripts/install_store.sh | sh

Python

.. code-block:: bash

    python -c 'import urllib; print urllib.urlopen("https://github.com/django-leonardo/django-leonardo/raw/develop/contrib/scripts/install_leonardo_dev.sh").read()' > install_leonardo.sh
    sudo sh install_leonardo.sh

Command by command

.. code-block:: bash

    virtualenv -p /usr/bin/python2.7 leonardo_venv
    cd leonardo_venv
    . $PWD/bin/activate

    pip install -e git+https://github.com/django-leonardo/django-leonardo@develop#egg=django-leonardo
    pip install -r $PWD/src/django-leonardo/requirements.txt
    django-admin startproject --template=https://github.com/django-leonardo/site-template/archive/master.zip myproject

    export PYTHONPATH=$PWD/myproject
    cd myproject

    python manage.py makemigrations --noinput
    python manage.py migrate --noinput
    python manage.py bootstrap_site --url=http://raw.githubusercontent.com/django-leonardo/django-leonardo/develop/contrib/bootstrap/demo.yaml

    echo "from django.contrib.auth.models import User; User.objects.create_superuser('root', 'mail@leonardo.cz', 'admin')" | python manage.py shell

    python manage.py runserver 0.0.0.0:80


Navigate your browser to your_ip/admin and login with ``root:admin``
For settings production mode could take inspiration from `Leonardo Documentation`_.

.. _`Leonardo Documentation`: http://django-leonardo.readthedocs.org/en/develop/install/production.html

Core
====

Leonardo in default state has enabled some modules which provides basic stuff for common sites:

* Web - precise FeinCMS integration
* Navigation - common navigation components
* Media - Filer integration with media widgets
* Search - Haystack integration
* Auth - Basic auth with standard user actions
* Devel - Widgets for site developers

Installed
=========

These modules are included in default leonardo installation, but could be uninstalled anytime without affecting your DB:

* System - Common management stuff (listing installed packages, widgets version etc..)
* Sitestarter - simple site starter which handle missing site and create it from custom yaml template
* Admin - django admin
* Ckeditor - Default wysiwyg editor for now. Alternatives: Redactor,Summernote,Tinymce..

For uninstalling::

    pip uninstall leonardo-system
    pip uninstall leonardo-sitestarter

For switch wysiwyg::

    pip uninstall leonardo-ckeditor
    pip install leonardo-redactor

Extensions
==========

Leonardo in default state using module loader which allows you to easy installation of new packages.

All modules lives in `Package Index`_.

.. _`Package Index`: http://packages.leonardo.robotice.org/

Modules
-------

* `Admin`_ - Standard Django admin
* `Admin Honeypot`_ - Django admin honeypot
* `Bootstrap Admin`_ - Bootstrap Django admin
* `Material Admin`_ - Material Django admin
* `Analytics`_ -  Analytics service integration for Leonardo projects
* `Blog`_ - Elephant Blog integration
* `Celery`_ - Celery workers for Leonardo CMS
* `Multisite`_ - Multi site with security
* `Folio`_ - Portfolio app
* `Geo`_ - Some geolocation related widgets (Google maps,..)
* `Galleries`_ - Awesome galleries
* `Forms`_ - Stable Form Designer integration with Remote Forms
* `Store`_ - Oscar Ecommerce - FeinCMS integration
* `News`_ - Heavy FeinCMS News
* `Newswall`_ - Newswall
* `Links`_ - navigation helpers bundles as Leonardo module
* `Redactor`_ - A lightweight wysiwyg editor for Leonardo
* `Sentry`_ - end-user friendly error page
* `Page Permissions`_ - extend Page and provide new Navigation templates with permissions
* `Team`_ - team model and widgets
* `Oembed`_ - oembed objects with caching
* Static - client-side libraries like an AngularJS, React, D3.js, ..


.. _`Admin`: https://github.com/leonardo-modules/leonardo-admin
.. _`Admin Honeypot`: https://github.com/leonardo-modules/leonardo-admin-honeypot
.. _`Bootstrap Admin`: https://github.com/leonardo-modules/leonardo-bootstrap-admin
.. _`Material Admin`: https://github.com/leonardo-modules/leonardo-material-admin
.. _`Forms`: https://github.com/leonardo-modules/leonardo-module-forms
.. _`Blog`: https://github.com/leonardo-modules/leonardo-module-blog
.. _`Celery`: https://github.com/leonardo-modules/leonardo-celery
.. _`Multisite`: https://github.com/leonardo-modules/leonardo-multisite
.. _`Folio`: https://github.com/leonardo-modules/leonardo-module-folio
.. _`Geo`: https://github.com/leonardo-modules/leonardo-geo
.. _`Galleries`: https://github.com/leonardo-modules/leonardo-gallery
.. _`Store`: https://github.com/leonardo-modules/leonardo-store
.. _`News`: https://github.com/leonardo-modules/leonardo-module-news
.. _`Links`: https://github.com/leonardo-modules/leonardo-module-links
.. _`Redactor`: https://github.com/leonardo-modules/leonardo-module-redactor
.. _`Sentry`: https://github.com/leonardo-modules/leonardo-module-sentry
.. _`Page Permissions`: https://github.com/leonardo-modules/leonardo-module-pagepermissions
.. _`Analytics`: https://github.com/leonardo-modules/leonardo-module-analytics
.. _`Team`: https://github.com/leonardo-modules/leonardo-team
.. _`Oembed`: https://github.com/leonardo-modules/leonardo-oembed

Themes
------

* `Bootwatch`_
* `AdminLTE`_

.. _`Bootwatch`: https://github.com/leonardo-modules/leonardo-theme-bootswatch
.. _`AdminLTE`: https://github.com/leonardo-modules/leonardo-theme-adminlte

Cookiecutters
-------------

Our cookiecutters are a good start for creating new `Site`_, `Module`_ or `Theme`_.

* `Site`_
* `Module`_
* `Theme`_

.. _`Site`: https://github.com/django-leonardo/cookiecutter-site
.. _`Module`: https://github.com/django-leonardo/cookiecutter-module
.. _`Theme`: https://github.com/django-leonardo/cookiecutter-theme

Bundles
-------

Leonardo defines a group of bundles that can be used
to install Leonardo and the dependencies for a given feature.

You can specify these in your requirements or on the ``pip`` comand-line
by using brackets.  Multiple bundles can be specified by separating them by
commas.

.. code-block:: bash

    $ pip install "django-leonardo[multisite]"

    $ pip install "django-leonardo[blog,store,multisite]"

`Here`_ is current list of bundles.

.. _`Here`: https://github.com/django-leonardo/django-leonardo/blob/master/setup.cfg#L28

Looking for commercial support?
===============================

If you are interested in having an Leonardo project built for you, or for development of an existing Leonardo site. Please get in touch via mail@majklk.cz.

Read More
=========

* http://docs.openstack.org/developer/horizon/quickstart.html
* http://feincms-django-cms.readthedocs.org/en/latest/index.html
* https://django-oscar.readthedocs.org/en/releases-1.0/

.. |Doc badge| image:: https://readthedocs.org/projects/django-leonardo/badge/?version=develop
.. |Pypi| image:: https://img.shields.io/pypi/dm/django-leonardo.svg?style=flat
.. |PypiVersion| image:: https://badge.fury.io/py/django-leonardo.svg?style=flat
.. |Travis| image:: https://travis-ci.org/django-leonardo/django-leonardo.svg?branch=master


ChangeLog
CHANGES
=======


1.0.5
-----

* Change all imports to string notation and move imports from widget to models. related to Django 1.9 permited import models before full setup
* Better info about errors in modules
* Test all widget classes
* Construct widget form just once
* Fix return in load_widget_classes
* First integration of python-app-loader, cleanup useless stuff
* Load widget classes in get_all_widgets method
* Register new content types in models.py and support strings and tuples. #122 From this time use strings instead of classes. Add www prefix to domain in readme
* Implement target blank option in Simple ImageWidget #123 and mark requirements #125
* Remove restframework relics
* Add new Django import path for generic FK and make logo optional
* Fix travis conf
* Dj 1.10 to allow failures
* Add 1.10 to allow failures
* Add widget dimensions table to widget create form. Which allows create dimensions during widget create
* Fix double requirement in travis tests
* Add Django 1.7 back to tests
* Drops django 1.7 from tests and add 1.10. We still support Django 1.7, but mptt 0.8 drops support for < 1.8
* Fix imports in new version of polymorphic
* Bump required version of django-filer. Because dependency to django-polymorphic brokes release
* Add support for custom constance fields. See django-constance repo for more details
* Fix delete widget success redirect and update links in README
* Remove ajax-modal class from target=_blank
* Tune frontend page admin
* Split ActionMixin and apply successurl to delete widget view
* Add widget copy action and cleanup modal headers and redirects
* Fix unicode on python 2
* Add common detail tab template
* Add pbr depndency. Which is included only in setup.py
* Add image field to Site Heading Widget
* Set version and changelog update
* cleanup tox
* Add search backend deps to travis tests
* Handle missing haystack backend
* small cleanup
* small docker fixes and change path to standard
* Get favicon key from constance directly may cause some error when this key is not present in settings
* freeze on compress 1.5
* add initial support for loading django apps #120
* adjust to compress 1.6
* make whoosh independent
* Add initial of Debian and Docker spec
* fix missing whoosh
* do not duplicate requirements in test env
* update requirements related to #118
* cleanup dependencies #118
* update version of django extensions because failing on dj 1.8.6 +
* improve handle uploaded files
* handle missing WOW js
* doc update and include django forms to leonardo forms
* expose MEDIA FILES and handle_uploaded_files function
* update whoosh backend
* remove useless common fragments

1.0.4
-----

* readme update
* include video widget
* delete cookie in unauth mode
* ug
* fix calling parent
* list mixin separation for using without model
* change collectstatic command verbosity and refactoring syncing tools #108
* onload svg viewBox
* add .embed-responsive from bootstrap 4 and restrict django versions
* reload on close iframe modal, use popup mode only for get requests
* fix urls
* spell fix
* adjust labels in select2 fields
* show edit button only in administrator mode
* use default decorator and make decorate_urlconf public method
* fix file admin common urls
* Loading spinner tweaked
* ckeditor to default requirements // temporary
* allow declaring form size directly from widget class or form
* cosmetic changes
* update required version
* move site bootstraping to leonardo-sitestarter module
* uf fix missing compressor
* cleanup html widget and add ability to override widgets on every feincms widget without creating form
* allow use already compressed js files
* better developers message
* disable quickstart middleware in production mode
* fix dimensions
* fix site logo
* loading style fix
* loading cleanup
* client side fixes
* ckeditor to default requirements // temporary
* allow declaring form size directly from widget class or form
* cosmetic changes
* update required version
* move site bootstraping to leonardo-sitestarter module
* uf fix missing compressor
* cleanup html widget and add ability to override widgets on every feincms widget without creating form
* allow use already compressed js files
* fix image url without defined size
* fix unfiled images has not attribute quoted_logical_path
* show override url field
* fix template loading
* switch license to BSD
* make site quickstart db error resistent
* fix dimensions
* fix site logo
* loading style fix
* loading cleanup
* client side fixes
* better developers message
* wrap_all removed
* disable quickstart middleware in production mode
* fix image url without defined size
* fix unfiled images has not attribute quoted_logical_path
* show override url field
* fix template loading
* switch license to BSD
* make site quickstart db error resistent
* fix dimensions
* fix site logo
* loading style fix
* loading cleanup
* client side fixes
* update release meta and no compress on init stuff

1.0.3
-----

* loading tunning
* modal size fix
* hide carousel backround
* loading tunning
* modal size fix
* hide carousel backround
* compress init stuff in one bundle
* fix media pagination
* hide duplicate asterisk
* add dbtemplate loader to debug mode
* fullscreen modal
* declare ngFit
* ignore non iterable modules
* remove profiling from default panels
* simple iframe admin like a django-cms
* fix matching other document types
* Compress seems to break modals - removed
* update changelog

1.0.2
-----

* no compress
* admin action and target all admin actions to blank window
* media quick actions and add tooltip for actions
* fix sqlite hack on python3
* fix svg scale
* fix svg / vector widget
* fix JS boolean evaluation
* use empty choice for layout in mass update and fix side bar
* compress and fix side bar
* small ui improvements
* use aplha version
* update requirements
* update django-select2 in test environment
* switch dbtemplates and constance to our builds
* fix remember checkbox

hrcms/models.py
# -#- coding: utf-8 -#-
from functools import update_wrapper

import six
from admin_tools.menu import items, Menu
from django import http, template
from django.conf import settings
from django.contrib import admin
from django.contrib.admin.sites import AdminSite
from django.core.urlresolvers import reverse
from django.db import models
from django.db.models.signals import pre_save
from django.shortcuts import render_to_response
from django.template import Context, loader, RequestContext
from django.template.loader import render_to_string
from django.utils.translation import ugettext_lazy as _
from django.views.decorators.cache import never_cache
from django_assets import Bundle, register
from django_extensions.db.fields.json import JSONField
from feincms.module.page.models import BasePage as FeinCMSPage
from feincms.models import Base as FeinCMSBase

from .const import *


class Page(FeinCMSPage):

    col1_width = models.IntegerField(verbose_name=_("Column 1 width"),
        choices=COLUMN_CHOICES, default=DEFAULT_WIDTH)
    col2_width = models.IntegerField(verbose_name=_("Column 2 width"),
        choices=COLUMN_CHOICES, default=DEFAULT_WIDTH)
    col3_width = models.IntegerField(verbose_name=_("Column 3 width"),
        choices=COLUMN_CHOICES, default=DEFAULT_WIDTH)
    col4_width = models.IntegerField(verbose_name=_("Column 4 width"),
        choices=COLUMN_CHOICES, default=DEFAULT_WIDTH)
    template_name = models.CharField(
        verbose_name="Template", max_length=255,
        help_text=_("Core HTML templates and CSS styles."), null=True, blank=True)
    theme = models.CharField(verbose_name=_("Theme"), max_length=255, null=True, blank=True,
                             help_text=_("Color and style extension to the template."))

    class Meta:
        verbose_name = _("Page")
        verbose_name_plural = _("Pages")
        ordering = ['tree_id', 'lft']


class Widget(FeinCMSBase):
    options = JSONField(
        verbose_name=_('widget options'), blank=True, editable=False)
    prerendered_content = models.TextField(
        _('prerendered content'), blank=True, editable=False)

    label = models.CharField(verbose_name=_("Title"), max_length=255)
    template_name = models.CharField(
        verbose_name=_("Display"), max_length=255)

    span = models.IntegerField(verbose_name=_("Span"),
        choices=COLUMN_CHOICES, default=DEFAULT_WIDTH)
    vertical_span = models.IntegerField(verbose_name=_("V. Span"),
        choices=ROW_CHOICES, default=DEFAULT_WIDTH)
    align = models.IntegerField(verbose_name=_("Alignment"), choices=ALIGN_CHOICES)
    vertical_align = models.IntegerField(
        verbose_name=_("V. Alignment"), choices=VERTICAL_ALIGN_CHOICES, default=DEFAULT_CHOICE)
    prepend = models.IntegerField(verbose_name=_("Prepend"),
        choices=COLUMN_CHOICES, default=DEFAULT_WIDTH)
    append = models.IntegerField(verbose_name=_("Append"),
        choices=COLUMN_CHOICES, default=DEFAULT_CHOICE)
    push = models.IntegerField(verbose_name=_("Push"),
        choices=COLUMN_CHOICES, default=DEFAULT_CHOICE)
    pull = models.IntegerField(verbose_name=_("Pull"),
        choices=COLUMN_CHOICES, default=DEFAULT_WIDTH)
    vertical_prepend = models.IntegerField(
        verbose_name=_("V. Prepend"), choices=ROW_CHOICES, default=DEFAULT_CHOICE)
    vertical_append = models.IntegerField(
        verbose_name=_("V. Append"), choices=ROW_CHOICES, default=DEFAULT_CHOICE)
    vertical_push = models.IntegerField(
        verbose_name=_("V. Push"), choices=ROW_CHOICES, default=DEFAULT_CHOICE)
    vertical_pull = models.IntegerField(verbose_name=_("V. Pull"),
        choices=ROW_CHOICES, default=DEFAULT_CHOICE)
    style = models.IntegerField(
        verbose_name=_("Style"), choices=STYLE_CHOICES, default=DEFAULT_CHOICE)
    border = models.IntegerField(
        verbose_name=_("Border"), choices=BORDER_CHOICES, default=DEFAULT_CHOICE)
    clear = models.IntegerField(
        verbose_name=_("Clear"), choices=CLEAR_CHOICES, default=DEFAULT_CHOICE)
    last = models.NullBooleanField(verbose_name=_('Is last?'))
    visible = models.NullBooleanField(verbose_name=_('Is visible?'))

    class Meta:
        abstract = True
        verbose_name = _("abstract widget")
        verbose_name_plural = _("abstract widgets")

    def thumb_geom(self):
        return config_value('MEDIA', 'THUMB_MEDIUM_GEOM')

    def thumb_options(self):
        return config_value('MEDIA', 'THUMB_MEDIUM_OPTIONS')

    def _render_box_classes(self):
        classes = []
        options = self.options
        try:
            if options['align'][0] != 'a':
                classes.append('align-%s' % options['align'][0])
            if options['align'][1] != 'a':
                classes.append('valign-%s' % options['align'][1])
            if options['size'][0] > 0:
                classes.append('span-%s' % options['size'][0])
            if options['size'][1] > 0:
                classes.append('vspan-%s' % options['size'][1])
            if options['padding'][0] > 0:
                classes.append('vprepend-%s' % options['padding'][0])
            if options['padding'][1] > 0:
                classes.append('append-%s' % options['padding'][1])
            if options['padding'][2] > 0:
                classes.append('vappend-%s' % options['padding'][2])
            if options['padding'][3] > 0:
                classes.append('prepend-%s' % options['padding'][3])
            if options['margin'][0] > 0:
                classes.append('vpull-%s' % options['margin'][0])
            if options['margin'][1] > 0:
                classes.append('push-%s' % options['margin'][1])
            if options['margin'][2] > 0:
                classes.append('vpush-%s' % options['margin'][2])
            if options['margin'][3] > 0:
                classes.append('pull-%s' % options['margin'][3])
            if options['clear'] == 'l':
                classes.append('clearfix')
            if options['clear'] == 'f':
                classes.append('clear')
            if options['border'] == '1':
                classes.append('border')
            if options['border'] == '2':
                classes.append('colborder')
            if options['style'] == '' or options['style'] == None:
                classes.append('normal-widget')
            else:
                classes.append('%s-widget' % options['style'])
            if options['last']:
                classes.append('last')
        except:
            classes.append('widget-error')
        return u' '.join(classes)
    render_box_classes = property(_render_box_classes)

    def _template_name(self):
        try:
            if self.options['template_name'] == '':
                template = 'default'
            else:
                template = self.options['template_name']
        except:
            template = 'default'
        return u'widget/%s/%s.html' % (self.widget_name, template)
    template_name = property(_template_name)

    def get_template_name(self, format):
        try:
            if self.options['template_name'] == '':
                template = 'default'
            else:
                template = self.options['template_name']
        except:
            template = 'default'
        return u'widget/%s/%s.%s' % (self.widget_name, template, format)
    template_name = property(_template_name)

    def _template_xml_name(self):
        template = 'default'
        return u'widget/%s/%s.xml' % (self.widget_name, template)
    template_xml_name = property(_template_xml_name)

    def _widget_name(self):
        return self.__class__.__name__.lower().replace('widget', '')
    widget_name = property(_widget_name)

    def _widget_label(self):
        return self._meta.verbose_name
    widget_label = property(_widget_label)

    def render(self, **kwargs):
        return self.render_content(kwargs)
        if self.prerendered_content == '':
            return self.render_content(kwargs)
        else:
            return self.prerendered_content

    def render_content(self, options):
        #        if options.has_key('use_xml') and options['use_xml']:
        #            template = loader.get_template(self.template_xml_name)
        if options.has_key('format'):
            format = options.get('format')
            base_template = 'widget/doc_base.%s' % format

        else:
            format = 'html'
            base_template = 'widget/base.%s' % format

        template = loader.get_template(self.get_template_name(format))

        context = RequestContext(options['request'], {
            'widget': self,
            'base_template': base_template,
            'request': options['request'],
        })
        return template.render(context)

    def render_error(self, error_code):
        return render_to_string("widget/error.html", {
            'widget': self,
            'request': kwargs['request'],
        })

    def save(self, *args, **kwargs):
        super(Widget, self).save(*args, **kwargs)
        if self.options == {}:
            self.options = DEFAULT_DISPLAY_OPTIONS
            self.save()
"""
jquery_js = Bundle(
    'lib/jquery/1.6/js/jquery.js', # 1.6.2
    'lib/jquery/1.4/js/hoverIntent.js', # r6
    'lib/jquery/1.4/js/notice.js', # 1.0.1
    'lib/jquery/1.4/js/cookie.js',
    'lib/jquery/1.4/js/form.js',
    'lib/jquery/1.4/js/blockUI.js', # 2.37
    'lib/jquery/1.4/js/hslides.js', # 1.0
    'lib/jquery/1.4/js/cycle.all.js', # 2.97
    'lib/jquery/1.6/js/color.js',
    'lib/jquery/1.4/js/timers.js',
    'lib/jquery/1.4/js/easing.js',
    'lib/jquery/1.4/js/jcarousel.js', # 0.2.8
    'lib/jquery/1.4/js/lavalamp.js',
    'lib/jquery/1.4/js/panelgallery.js',
    'lib/jquery/1.4/js/dajax.core.js',
    'lib/jquery/1.4/js/colorbox.js',
    'lib/jquery/1.4/js/rating.js', # 3.13
    'lib/jquery/1.4/js/superfish.js', # 1.4.8
    'lib/jquery/1.4/js/supersubs.js', # 0.2b
    'lib/swfobject/swfobject.js', # 2011-02-23
    'lib/utils/json2.js', # 2011-02-23
    'lib/utils/DD_belatedPNG.js', # 0.0.8a
    filters='jsmin',
    output='_cache/jquery.js'
)
register('jquery_js', jquery_js)

jquery_ui_js = Bundle(
    # core
    'lib/jquery-ui/1.9/js/ui.core.js',
    'lib/jquery-ui/1.9/js/ui.widget.js',
    'lib/jquery-ui/1.9/js/ui.mouse.js',
    'lib/jquery-ui/1.9/js/ui.position.js',
    'lib/jquery-ui/1.9/js/ui.draggable.js',
    'lib/jquery-ui/1.9/js/ui.sortable.js',
    'lib/jquery-ui/1.9/js/ui.droppable.js',
    'lib/jquery-ui/1.9/js/ui.resizable.js',
    # widgets
    'lib/jquery-ui/1.9/js/ui.button.js',
    'lib/jquery-ui/1.9/js/ui.dialog.js',
    'lib/jquery-ui/1.9/js/ui.slider.js',
    'lib/jquery-ui/1.9/js/ui.tabs.js',
    'lib/jquery-ui/1.9/js/ui.autocomplete.js',
    'lib/jquery-ui/1.9/js/ui.datepicker.js',
    'lib/jquery-ui/1.9/js/ui.tooltip.js',
    filters='jsmin',
    output='_cache/jquery_ui.js'
)
register('jquery_ui_js', jquery_ui_js)

jquery_ui_css = Bundle(
    'lib/jquery-ui-themes/absolution/jquery.ui.base.css',
    'lib/jquery-ui-themes/absolution/jquery.ui.core.css',
    'lib/jquery-ui-themes/absolution/jquery.ui.resizable.css',
    'lib/jquery-ui-themes/absolution/jquery.ui.selectable.css',
    'lib/jquery-ui-themes/absolution/jquery.ui.autocomplete.css',
    'lib/jquery-ui-themes/absolution/jquery.ui.button.css',
    'lib/jquery-ui-themes/absolution/jquery.ui.dialog.css',
    'lib/jquery-ui-themes/absolution/jquery.ui.tabs.css',
    'lib/jquery-ui-themes/absolution/jquery.ui.slider.css',
    'lib/jquery-ui-themes/absolution/jquery.ui.datepicker.css',
    'lib/jquery-ui-themes/absolution/jquery.ui.tooltip.css',
    'lib/jquery-ui-themes/absolution/colors/jquery.ui.colors.blue.css',
    filters='cssmin,cssrewrite',
    output='_cache/jquery_ui.css'
)
register('jquery_ui_css', jquery_ui_css)

blueprint_css = Bundle(
    'lib/blueprint/1.0/css/screen.css',
    'lib/blueprint/1.0/css/plugins/vertical-positioning/screen.css',
    'lib/blueprint/1.0/css/plugins/liquid/liquid.css',
    filters='cssmin,cssrewrite',
    output='_cache/blueprint.css'
)
register('blueprint_css', blueprint_css)

webcms_js = Bundle(
    'lib/webcms/js/forms.js',
    filters='jsmin',
    output='_cache/webcms.js'
)
register('webcms_js', webcms_js)

webcms_css = Bundle(
    'lib/webcms/css/forms.css',
    'lib/webcms/css/highlight.css',
    'lib/jquery/1.4/css/rating.css',
    filters='cssmin,cssrewrite',
    output='_cache/webcms.css'
)
register('webcms_css', webcms_css)

webcms_fe_js = Bundle(
    'lib/jquery/1.4/js/stylesheetToggle.js',
    'lib/jquery/1.6/js/griddy.js',
    'lib/jstree/1.0/jstree.js',
    'lib/webcms/js/frontend_editing.js',
    filters='jsmin',
    output='_cache/webcms_fe.js'
)
register('webcms_fe_js', webcms_fe_js)

webcms_fe_css = Bundle(
    'lib/webcms/css/frontend_editing.css',
    'lib/jstree/1.0/themes/default/style.css',
    filters='cssmin,cssrewrite',
    output='_cache/webcms_fe.css'
)
register('webcms_fe_css', webcms_fe_css)

webcms_admin_js = Bundle(
    'lib/tinymce/3.4/tiny_mce.js',
    'lib/tinymce/3.4/jquery.tinymce.js',
    filters='jsmin',
    output='_cache/webcms_admin.js'
)
register('webcms_admin_js', webcms_admin_js)

webcms_admin_css = Bundle(
    'lib/codemirror2/codemirror.css',
    filters='cssmin,cssrewrite',
    output='_cache/webcms_admin.css'
)
register('webcms_admin_css', webcms_admin_css)
"""
